packager/mpd/base/period.cc

```diff
@@ -220,11 +232,16 @@ bool Period::SetNewAdaptationSetAttributes(
           FindOriginalAdaptationSetForTrickPlay(
               media_info, content_protection_in_adaptation_set);
       if (!trick_play_reference_adaptation_set) {
-        LOG(ERROR) << "Failed to find original AdaptationSet for trick play.";
-        return false;
+        MediaInfo media_info_no_trickplay = media_info;
+        media_info_no_trickplay.mutable_video_info()->clear_playback_rate();
+        std::string key =
+            GetAdaptationSetKey(media_info_no_trickplay,
+                                mpd_options_.mpd_params.allow_codec_switching);
```

Comment:
Suggest creating a private function in Period to avoid code duplication:

```c++
std::string Period::GetAdaptationSetKeyForTrickPlay(const MediaInfo& media_info) {
  MediaInfo media_info_no_trickplay = media_info;
  media_info_no_trickplay.mutable_video_info()->clear_playback_rate();
  return GetAdaptationSetKey(media_info_no_trickplay,
                             mpd_options_.mpd_params.allow_codec_switching);
}
```